### ISSUE TITLE
Training, validation and test

### ISSUE THREAD

**Issue by perecanals** [MIC-DKFZ/nnUNet#167]
Hi Fabian,

I have some doubts regarding how the algorithm divides data into training set, validation set and test set. I initially assumed that images and labels from from images/labelsTr folders were training data and images from imagesTs were used for validation, and then a separate dataset should be used for test. Is that how it works?

If it isn't, I understand that training+validation are all from images/labelsTr, and then test images are the ones in imagesTs. But then, how do I get test values from the test dataset?

What I am currently doing is putting about 60% of my data in the imagesTr/labelsTr, 20% in the imagesTs and saving the 20% left for testing that I perform on my own by inferencing and comparing the output with the labels, and I am performing cross-validation by shuffling iteratively the dataset with these proportions. Am I doing it correctly? 

I am quite positive I am not entirely understanding how the net manages data in this sense. 

Thank you in advance,

**Comment by FabianIsensee** [comment]
nnunet runs 5 fold cross-validation on the training data. Usually in challenges you dont have the test labels so you need to submit the test set to some website

**Comment by perecanals** [comment]
Thank you for your quick answer. So then, if I test my code separately, I needn't put any images in the folder imagesTs?

I am a bit worried about this because at the moment I have a very low amount of data, and I would like for the network to perform the testing on unseen data. If I select the option ```all``` for ```FOLD``` in the following line:

``` python run/run_training.py 3d_fullres nnUNetTrainer TaskXX_MY_DATASET FOLD --ndet ```

Will it train with all the samples in the imagesTr dir? If, alternatively, I used 4 (I understand it would be the 5-fold cross validation), how would the algorithm treat my data? Would it use part of the training set for each fold?

Thank you again.

**Comment by FabianIsensee** [comment]
Hi,
you don't need to have anything in imagesTs and nnU-Net does not use/need this data for anything. imagesTs is there because this is the way the Decathlon used to do it. If you want predictions on your test images you need to manually trigger that.
I wholeheartedly agree: you need to test on completely separate data. If you place your test cases in imagesTs (or any other folder) nnU-Net will not touch them at all.
I do not necessarily recomment running fold=all. This will not give you a performance estimate on your training data. It is better to train all 5 folds because this gives you said estimate but also gives you 5 models to be used in an ensemble. Especially for small datasets this can give a nice boost in performance.
5 fold cross-validation takes all cases in imagesTr, splits them into 5 equally sized parts (80:20) and always trains on 4/5 parts while evaluating on 1/5. 
Best,
Fabian
